app/services/streaming/streaming.ts

```diff
@@ -372,6 +421,21 @@ export class StreamingService
     }
   }
 
+  private handleSetupPlatformError(e: unknown, platform: TPlatform) {
+    console.error('Error running beforeGoLive for platform', e);
+    // cast all PLATFORM_REQUEST_FAILED errors to SETTINGS_UPDATE_FAILED
+    if (e instanceof StreamError) {
+      e.type =
+        (e.type as TStreamErrorType) === 'PLATFORM_REQUEST_FAILED'
+          ? 'SETTINGS_UPDATE_FAILED'
+          : e.type || 'UNKNOWN_ERROR';
+      this.setError(e, platform);
```

Comment:
why are we doing this?

Reply:
This is a copy/paste from the previous code, which I separated out into its own function to make it dry when I call it in the Green code. The error types determine which error message is shown to the user.